core/kernel/wait_queue.c

```diff
@@ -69,24 +75,47 @@ void wq_wait_init_condvar(struct wait_queue *wq, struct wait_queue_elem *wqe,
 	cpu_spin_unlock_xrestore(&wq_spin_lock, old_itr_status);
 }
 
-void wq_wait_final(struct wait_queue *wq, struct wait_queue_elem *wqe,
-		   const void *sync_obj, const char *fname, int lineno)
+static uint32_t wq_wait_final_helper(struct wait_queue *wq,
+				     struct wait_queue_elem *wqe,
+				     uint32_t timeout_ms, const void *sync_obj,
+				     const char *fname, int lineno)
 {
 	uint32_t old_itr_status;
 	unsigned done;
+	uint32_t ret;
```

Comment:
`TEE_Result`
Please initialize all the local variables in this function since you're adding a local variable.